craft_parts/executor/executor.py

```diff
@@ -171,7 +171,7 @@ def _install_build_packages(self) -> None:
         if self._extra_build_packages:
             build_packages.update(self._extra_build_packages)
 
-        packages.Repository.install_build_packages(sorted(build_packages))
+        packages.Repository.install_packages(sorted(build_packages))
```

Comment:
Renamed because it's not restricted to build packages anymore, it can also be used to install overlay packages.